packages/task/src/browser/task-service.ts

```diff
@@ -293,8 +311,57 @@ export class TaskService implements TaskConfigurationClient {
         this.logger.debug(`Task created. Task id: ${taskInfo.taskId}`);
 
         // open terminal widget if the task is based on a terminal process (type: shell)
+        // or attach to process output processor if a raw process (type: process)
```

Comment:
I guess my biggest issue with this change is this part: both `process` and `shell` task start in a `TerminalProcess`. See https://github.com/theia-ide/theia/commit/2b950102fbd058459ebe227b9e128e3637bc8ad2.

Maybe for background tasks we may want to spawn them in a `RawProcess`, since it will never be shown to a user... But at the time I wrote the task process spawning I don't think it was implemented. IIRC when looking at VS Code sources, they used to always spawn everything in a `TerminalProcess`-like fashion as well.

Here are the characteristics of the two kind of processes in Theia:
- `RawProcess`: use the regular spawn/fork methods, you can directly hook into stdin/stdout/stderr.
- `TerminalProcess`: probably a confusing name, but all it does is create a pty/console _-not the widget-_ and spawns the process inside it, you only get back stdin/stdout (stderr is "mixed" with the process stdout when it comes out of the console/pseudo-terminal).

I wasn't able to test your change in the branches that expect `RawProcess` instances, since we never create tasks using them (as of now). I assume that in your custom task runner, you do create `RawProcess` instances, but I feel like this should be tested somehow.

Reply:
I have removed the test altogether so now RawProcess and TerminalProcess are treated exactly the same.  For some reason I thought they needed to be treated differently but in fact all the API needed was in the base Process.